spec/item_spec.rb

```diff
@@ -47,6 +56,23 @@
       model = Item.new
       expect { Organization.as_current(model) {}}.to raise_error("#{model} is not a Organization")
     end
+
+    it 'invalidates dependent models' do
+      DependentModel.current = DependentModel.create!
+      dependent_id = DependentModel.current.object_id
+
+      SubOrganization.create!.as_current do
+        expect(DependentModel.current.object_id).not_to eq(dependent_id)
```

Comment:
Minor but you can use [equal](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/equality-matchers#compare-using-equal-(equal?)) matcher to compare object identity rather than relying on `object_id`.